ctapipe/reco/sklearn.py

```diff
@@ -507,6 +501,318 @@ def predict_table(self, key, table: Table) -> Table:
+                    fov_lat = hillas.fov_lat + disp * np.sin(psi)
+                    alt, az = telescope_to_horizontal(
+                        lon=fov_lon,
+                        lat=fov_lat,
+                        pointing_alt=event.pointing.tel[tel_id].altitude.to(u.deg),
+                        pointing_az=event.pointing.tel[tel_id].azimuth.to(u.deg),
+                    )
+
+                    altaz_container = ReconstructedGeometryContainer(
+                        alt=alt[0], az=az[0], is_valid=True
+                    )
+
+                else:
+                    altaz_container = ReconstructedGeometryContainer(
+                        alt=u.Quantity(np.nan, u.deg, copy=False),
+                        az=u.Quantity(np.nan, u.deg, copy=False),
+                        is_valid=False,
+                    )
+            else:
+                disp_container = DispContainer(
+                    norm=u.Quantity(np.nan, self.unit),
+                    is_valid=False,
+                )
+                altaz_container = ReconstructedGeometryContainer(
+                    alt=u.Quantity(np.nan, u.deg, copy=False),
+                    az=u.Quantity(np.nan, u.deg, copy=False),
+                    is_valid=False,
+                )
+
+            disp_container.prefix = f"{self.prefix}_parameter"
+            altaz_container.prefix = f"{self.prefix}_tel"
+            event.dl2.tel[tel_id].disp[self.prefix] = disp_container
+            event.dl2.tel[tel_id].geometry[self.prefix] = altaz_container
+
+        self.stereo_combiner(event)
+
+    def predict_table(self, key, table: Table) -> Tuple[Table, Table]:
+        """Predict on a table of events
+
+        Parameters
+        ----------
+        table : `~astropy.table.Table`
+            Table of features
+
+        Returns
+        -------
+        disp_table : `~astropy.table.Table`
+            Table with disp predictions, matches the corresponding
+            container definition
+        altaz_table : `~astropy.table.Table`
+            Table with resulting predictions of horizontal coordinates
+        """
+        table = self.feature_generator(table)
+
+        n_rows = len(table)
+        disp = u.Quantity(np.full(n_rows, np.nan), self.unit, copy=False)
+        is_valid = np.full(n_rows, False)
+
+        valid = self.quality_query.get_table_mask(table)
+        disp[valid], is_valid[valid] = self._predict(key, table[valid])
+
+        disp_result = Table(
+            {
+                f"{self.prefix}_parameter_norm": disp,
+                f"{self.prefix}_parameter_is_valid": is_valid,
+            }
+        )
+        add_defaults_and_meta(
+            disp_result,
+            DispContainer,
+            prefix=f"{self.prefix}_parameter",
+            stereo=False,
+        )
+
+        psi = table["hillas_psi"].quantity.to_value(u.rad)
+        fov_lon = table["hillas_fov_lon"] + disp * np.cos(psi)
```

Comment:
This should be `table["hillas_fov_lon"].quantity`, right?

Reply:
Yes, fixed.